build-info-extractor/src/main/java/org/jfrog/build/extractor/clientConfiguration/client/ArtifactoryXrayClient.java

```diff
@@ -44,11 +44,15 @@
      */
     private static final int HTTP_CLIENT_RETRIES = 0;
 
-    public ArtifactoryXrayClient(String artifactoryUrl, String username, String password, Log logger) {
-        super(artifactoryUrl, username, password, StringUtils.EMPTY, logger);
+    public ArtifactoryXrayClient(String artifactoryUrl, String username, String password, String accessToken, Log logger) {
```

Comment:
Don’t we need a constructor that received only access token? Why do we need to pass username, password and accessToken?

Reply:
This Base client contains the following logic:
```java
if (StringUtils.isNotEmpty(accessToken)) {
    httpClient = new ArtifactoryHttpClient(this.artifactoryUrl, accessToken, logger);
} else {
    httpClient = new ArtifactoryHttpClient(this.artifactoryUrl, username, password, logger);
}
```
With that, we don't have to add the above check in the Jenkins plugin. For example, in the Jenkins plugin I did this:

```java
ArtifactoryXrayClient client = new ArtifactoryXrayClient(server.getUrl(), credentials.getUsername(), credentials.getPassword(), credentials.getAccessToken(), log);

```

Reply:
Got it. OK.